src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup/Cmdlets/Restore/RestoreAzureRMRecoveryServicesBackupItem.cs

```diff
@@ -88,27 +150,49 @@ public override void ExecuteCmdlet()
+                {
+                    GenericResource targetStorageAccountResource = GetStorageAccountResource(TargetStorageAccountName,
+                        TargetStorageAccountResourceGroupName);
+                    WriteDebug(string.Format("StorageId = {0}", targetStorageAccountResource.Id));
+                    providerParameters.Add(RestoreFSBackupItemParams.TargetStorageAccountId,
+                        targetStorageAccountResource.Id);
+                    providerParameters.Add(RestoreFSBackupItemParams.TargetStorageAccountLocation,
+                        targetStorageAccountResource.Location);
+                }
 
-                Dictionary<Enum, object> providerParameters = new Dictionary<Enum, object>();
                 providerParameters.Add(VaultParams.VaultName, vaultName);
                 providerParameters.Add(VaultParams.ResourceGroupName, resourceGroupName);
                 providerParameters.Add(VaultParams.VaultLocation, VaultLocation);
                 providerParameters.Add(RestoreBackupItemParams.RecoveryPoint, RecoveryPoint);
-                providerParameters.Add(RestoreBackupItemParams.StorageAccountId, storageAccountResource.Id);
-                providerParameters.Add(RestoreBackupItemParams.StorageAccountLocation, storageAccountResource.Location);
-                providerParameters.Add(RestoreBackupItemParams.StorageAccountType, storageAccountResource.Type);
-                providerParameters.Add(RestoreBackupItemParams.OsaOption, UseOriginalStorageAccount.IsPresent);
+                providerParameters.Add(RestoreVMBackupItemParams.OsaOption, UseOriginalStorageAccount.IsPresent);
+                providerParameters.Add(RestoreFSBackupItemParams.ResolveConflict, ResolveConflict.ToString());
+                providerParameters.Add(RestoreFSBackupItemParams.SourceFilePath, SourceFilePath);
+                providerParameters.Add(RestoreFSBackupItemParams.TargetStorageAccountName, TargetStorageAccountName);
+                providerParameters.Add(RestoreFSBackupItemParams.TargetFileShareName, TargetFileShareName);
+                providerParameters.Add(RestoreFSBackupItemParams.TargetFolder, TargetFolder);
 
                 if (TargetResourceGroupName != null)
                 {
-                    providerParameters.Add(RestoreBackupItemParams.TargetResourceGroupName, TargetResourceGroupName);
+                    providerParameters.Add(RestoreVMBackupItemParams.TargetResourceGroupName, TargetResourceGroupName);
+                }
+
+                if (SourceFileType != null)
+                {
+                    providerParameters.Add(RestoreFSBackupItemParams.SourceFileType, SourceFileType.ToString());
                 }
 
                 PsBackupProviderManager providerManager =
-                    new PsBackupProviderManager(providerParameters, ServiceClientAdapter);
+                        new PsBackupProviderManager(providerParameters, ServiceClientAdapter);
```

Comment:
Please do Ctrl + K + D on all the touched files to make sure all the files are in proper format. Also, do "remove and sort usings"